tests/countries/test_sao_tome_and_principe.py

```diff
@@ -0,0 +1,126 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS.md file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.sao_tome_and_principe import SaoTomeAndPrincipe
+from tests.common import CommonCountryTests
+
+
+class TestSaoTomeAndPrincipe(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(SaoTomeAndPrincipe, years=range(2014, 2050))
+        cls.observed_holidays = SaoTomeAndPrincipe(years=range(2014, 2050), observed=True)
+
+    def test_new_years_day(self):
+        for year in range(2014, 2050):
+            self.assertHolidayName("Ano Novo", f"{year}-01-01")
+
+    def test_king_amador_day(self):
+        for year in range(2014, 2050):
+            self.assertHolidayName("Dia do Rei Amador", f"{year}-01-04")
+
+    def test_martyrs_day(self):
+        for year in range(2014, 2050):
+            self.assertHolidayName("Dia dos Mártires", f"{year}-02-03")
+
+    def test_labor_day(self):
+        for year in range(2014, 2050):
+            self.assertHolidayName("Dia do Trabalhador", f"{year}-05-01")
+
+    def test_independence_day(self):
+        for year in range(2014, 2050):
+            self.assertHolidayName("Dia da Independência", f"{year}-07-12")
+
+    def test_armed_forces_day(self):
+        for year in range(2014, 2050):
+            self.assertHolidayName("Dia das Forças Armadas", f"{year}-09-06")
+
+    def test_agrarian_reform_day(self):
+        for year in range(2014, 2050):
+            self.assertHolidayName("Dia da Reforma Agrária", f"{year}-09-30")
+
+    def test_sao_tome_day(self):
+        for year in range(2019, 2050):  # Only from 2019 onward
+            self.assertHolidayName("Dia de São Tomé", f"{year}-12-21")
```

Comment:
For cases like this, we need to observe both before and after the holiday got add/removed.
```suggestion
        name = "Dia de São Tomé"
        self.assertHolidayName(name, (f"{year}-12-21" for year in range(2019, 2050)))
        self.assertNoHolidayName(name, range(2014, 2019))
```